promql/lex.go

```diff
@@ -123,6 +123,10 @@ func (i ItemType) isRightAssociative() bool {
 
 type ItemType int
 
+// This is a list of all keywords in PromQL
+// When changing this list, make sure to also change
+// the maybe_label grammar rule in the generated parser
+// to avoid misinterpretation of labels as keywords
```

Comment:
Full stop.

Reply:
Fixed it.

Given how often this happens to me, I maybe should write some script to check for this specific error automatically.